### DISCUSSION TITLE
Migration from one instance to another

### DISCUSSION THREAD

**Discussion by manjirisulekavadi** [Sunbird-Ed/Community#294] [General Discussion]
Is there a possibility of a smooth migration of an active tenant from one instance to another if required?

**Comment by Krishnaj20** [comment]
@manjirisulekavadi - can you specify more details about what do you want to migrate and why? This may help us understand the use case and respond accordingly.

@vinukumar-vs @rhwarrier

**Reply by manjirisulekavadi** [reply]
As of now hypothetical - but for example, Tata Engineering Services go for one single instance and then might want to move some subsidiaries to independent instance.

Or the State/Board content on Diksha needs to move to independent instance

**Reply by Krishnaj20** [reply]
I couldn't understand the DIKSHA use case. In DIKSHA all content is CC-BY licence which means the content discoverability and access is public by default.

In the other example, can you explain what do you mean by instance? Are you referring to tenant as instance? In the same use case it will be useful to know the scope of the migration. 

@rhwarrier @vrayulu @parthlawate

**Reply by parthlawate** [reply]
Let me give some more context. We have some Sunbird SAAS prospects who will likely use Sunbird as SAAS tenants to start with . but say a year down the line, they will want us to setup their SB infra on their own cloud. In this case we will need to migrate them from SAAS to their own SB Installation. 

Hope this explains

**Reply by Krishnaj20** [reply]
Got it. I believe migration scripts should address this scenario. Currently, I don't think anyone from the community has created these scripts yet.

@rhwarrier @vrayulu - tagging you just in case you may want to add more details here.

**Reply by parthlawate** [reply]
We have a live use case where one SAAS Adopter -- Which is a Subset of this larger migration use case..

There they want to migrate content & courses they have created in a Sunbird SAAS Tenant to Diksha & Vice versa..

**Reply by rhwarrier** [reply]
Moving a tenant lock, stock and barrel from one instance to another is something that hasn't been explored yet - there's a whole host of things that are tied to the tenant - Users, Orgs, content associations, usage attribution and more. We'll need to figure this out properly.

**Reply by Krishnaj20** [reply]
@parthlawate - We can seek contribution from the community to address this scenario. If this is urgent, we can assign necessary tech sponsors from Knowlg & Lern building blocks to help you out. 

@rhwarrier @vinukumar-vs @vrayulu
